clippy_lints/src/loops/manual_find.rs

```diff
@@ -35,6 +35,7 @@ pub(super) fn check<'tcx>(
         && let ExprKind::Call(ctor, [inner_ret]) = ret_value.kind
         && is_res_lang_ctor(cx, path_res(cx, ctor), LangItem::OptionSome)
         && path_res(cx, inner_ret) == Res::Local(binding_id)
+        && !contains_return(cond)
```

Comment:
Even though this would be some strangely-designed code, we should also protect against other diverging paths in the condition, such as `break` and `continue` such as in

```rust
pub fn first_less_than_10(v: Vec<u32>) -> Option<u32> {
    for x in v {
        if if x < 3 { break; } else { x < 10 } {
            return Some(x);
        }
    }
    None
}
```

```suggestion
        && !contains_return_break_continue_macro(cond)
```

Reply:
Thanks!